### ISSUE TITLE
Faster RCNN model version and Object Tag Sequences

### ISSUE THREAD

**Issue by xiaoleihuang** [microsoft/Oscar#13]
Did you use the open sourced version of the faster rcnn from torchvision: [https://pytorch.org/docs/stable/torchvision/models.html#torchvision.models.detection.fasterrcnn_resnet50_fpn](https://pytorch.org/docs/stable/torchvision/models.html#torchvision.models.detection.fasterrcnn_resnet50_fpn)?
And, did you use the open sourced version of tags and labels?

**Comment by xjli** [comment]
For vg object labels, we use the opensourced bottom-up attention repo (https://github.com/peteanderson80/bottom-up-attention).

**Comment by xiaoleihuang** [comment]
> For vg object labels, we use the opensourced bottom-up attention repo (https://github.com/peteanderson80/bottom-up-attention).

Did you directly apply the released pretrained faster rcnn model (https://www.dropbox.com/s/5xethd2nxa8qrnq/resnet101_faster_rcnn_final.caffemodel?dl=1) on the VQA images to get the semantic tags?

**Comment by xiaoleihuang** [comment]
The VQA uses the train2014_vg_qla_mrcnn as input, were the tags extracted by Masked R-CNN? But the paper says the tags were extracted by Faster R-CNN. .. Just want to confirm that.

**Comment by xjli** [comment]
There are two kinds of corpus in this work, pre-training corpus and downstream task finetuning corpus, both are using tag sets. For finetuning VQA, we observed that the best tag set is from COCO, not VG tag set (80 categories), because Faster R-CNN (pretrained on VG corpus), its tag prediction precision is not good enough, though VG tag set has more categories (1600). Here we use a high-precision Mask R-CNN (trained on COCO tag set) to generate the tags. For pre-training corpus, we use VG tag set from Faster R-CNN.

**Comment by xiaoleihuang** [comment]
Can you share the mrcnn model trained on COCO tagset?

**Comment by xjli** [comment]
I used torchvision.maskrcnn to generate the tags, 

model = torchvision.models.detection.maskrcnn_resnet50_fpn(pretrained=True)
maskrcnn = GeneralizedRCNN(model.backbone, model.rpn, model.roi_heads, model.transform)

It is opensourced for both API and pretrained models.

**Comment by xiaoleihuang** [comment]
I see. I thought you used the different versions of mrcnn. I have tried the open sourced one from torchvision. But I find that the tag set in `train2014_qla_mrcnn` is different from the open-sourced version tag set: there are 28 tags that exist in the json file but not found in the open-sourced categories. I tested the `train+val2014_qla_mrcnn` and the one with using vg. Their tag sets are different from the open-sourced version as well.

**Comment by xjli** [comment]
May try different confidence_threshold = 0.2 or 0.4 to generate (filter) the tags.

**Comment by xiaoleihuang** [comment]
The default category list is COCO_INSTANCE_CATEGORY_NAMES (https://pytorch.org/docs/stable/torchvision/models.html). Then I collected all the tag set from each json file aforementioned, I compare their differences, then I find 28 tags in the json files but not in the default category list.

**Comment by xjli** [comment]
See this, https://github.com/pytorch/vision/issues/990
please take your time to explore by yourself ...
